release-notes/7.0/preview/api-diff/rc1/Microsoft.NETCore.App/7.0-rc1_System.Runtime.Serialization.DataContracts.md

```diff
@@ -0,0 +1,55 @@
+# System.Runtime.Serialization.DataContracts
+
+``` diff
++namespace System.Runtime.Serialization.DataContracts {
++    public abstract class DataContract {
++        public virtual DataContract? BaseContract { get; }
++        public virtual string? ContractType { get; }
++        public virtual ReadOnlyCollection<DataMember> DataMembers { get; }
++        public virtual bool IsBuiltInDataContract { get; }
++        public virtual bool IsISerializable { get; }
++        public virtual bool IsReference { get; }
++        public virtual bool IsValueType { get; }
++        public virtual Dictionary<XmlQualifiedName, DataContract>? KnownDataContracts { get; }
++        public virtual Type OriginalUnderlyingType { get; }
++        public virtual XmlDictionaryString? TopLevelElementName { get; }
++        public virtual XmlDictionaryString? TopLevelElementNamespace { get; }
++        public virtual Type UnderlyingType { get; }
++        public virtual XmlQualifiedName XmlName { get; }
++        public virtual XmlQualifiedName GetArrayTypeName(bool isNullable);
++        public static DataContract? GetBuiltInDataContract(string name, string ns);
++        public static XmlQualifiedName GetXmlName(Type type);
++        public virtual bool IsDictionaryLike([NotNullWhenAttribute(true)] out string? keyName, [NotNullWhenAttribute(true)] out string? valueName, [NotNullWhenAttribute(true)] out string? itemName);
++    }
++    public sealed class DataContractSet {
++        public DataContractSet(DataContractSet dataContractSet);
++        public DataContractSet(ISerializationSurrogateProvider? dataContractSurrogate, IEnumerable<Type>? referencedTypes, IEnumerable<Type>? referencedCollectionTypes);
++        public Dictionary<XmlQualifiedName, DataContract> Contracts { get; }
++        public Dictionary<XmlQualifiedName, DataContract>? KnownTypesForObject { get; }
++        public Dictionary<DataContract, object> ProcessedContracts { get; }
++        public Hashtable SurrogateData { get; }
++        public DataContract GetDataContract(Type type);
++        public DataContract? GetDataContract(XmlQualifiedName key);
++        public Type? GetReferencedType(XmlQualifiedName xmlName, DataContract dataContract, out DataContract? referencedContract, out object[]? genericParameters, bool? supportGenericTypes = default(bool?));
```

Comment:
The code here actually defaults the `supportGenericTypes` parameter to `null`... which is the same as this default. But I see the other API's in this review just use "param = null" instead of "default()" for a nullable type. Not sure why this one came out differently.

Reply:
Could be a bug in the tool on how it detects the diff. If you think it needs fixing, please open a bug in dotnet/arcade for AsmDiff.